src/schema/v2/lot.ts

```diff
@@ -0,0 +1,179 @@
+import { connectionWithCursorInfo } from "./fields/pagination"
+import gql from "lib/gql"
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { connectionFromArray } from "graphql-relay"
+
+export const LotType = new GraphQLObjectType<any, ResolverContext>({
+  name: "Lot",
+  description:
+    "A lot in an auction containing merged Sale artwork and Lot state data",
+  interfaces: () => {
+    return [NodeInterface]
+  },
+  fields: () => {
+    return {
+      // Must place an `id` at the root of the object (next to saleArtwork)
+      // when resolving a lot for this to work (see resolver)
+      ...SlugAndInternalIDFields,
+      saleArtwork: {
+        type: SaleArtworkType,
+        description: "The saleArtwork object.",
+        resolve: ({ saleArtwork }) => {
+          return saleArtwork
+        },
+      },
+    }
+  },
+})
+
+/**
+ * Fetch relevant sale artworks and use them to fetch lot states
+ * from causality. Then pass the lot states through to the stitching layer
+ * via the graphql request context.
+ */
+const resolveWatchedLotConnection = async (_parent, args, context) => {
+  const { saleArtworksAllLoader, causalityLoader } = context
+  // fetch sale artworks from gravity
+  const { first = 25, ...rest } = args
+
+  const connectionOptions = {
+    include_watched_artworks: true,
+    total_count: true,
+    first,
+    ...rest,
+  }
+  const params = convertConnectionArgsToGravityArgs(connectionOptions)
+  delete params.page
+
+  const { body, headers } = await saleArtworksAllLoader(params)
+  const watchedSaleArtworks: any[] = body
+  const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+  const saleArtworkIds = watchedSaleArtworks.map((sa) => sa._id)
+
+  // Fetch the all fields of the lot object
+  const lotStatesResponse = await causalityLoader({
+    query: gql`
+      query WatchedLotsQuery($ids: [ID!]!) {
+        lots(ids: $ids) {
+          id
+          internalID
+          saleId
+          bidCount
+          reserveStatus
+          sellingPriceCents
+          onlineAskingPriceCents
+          floorSellingPriceCents
+          onlineSellingToBidder {
+            __typename
+            ... on ArtsyBidder {
+              id
+              paddleNumber
+              userId
+            }
+          }
+          floorWinningBidder {
+            __typename
+            ... on ArtsyBidder {
+              id
+              paddleNumber
+              userId
+            }
+          }
+          soldStatus
+        }
+      }
+    `,
+    variables: {
+      ids: saleArtworkIds,
+    },
+  }).then((res: FetchResponse) => res.json())
+
+  const { data: causalityData, errors: causalityErrors } = lotStatesResponse
+
+  // If the causality request failed for some reason, throw its errors.
+  if (causalityErrors) {
+    const errors = causalityErrors.reduce((acc, ce) => {
+      return acc + " " + ce["message"]
+    }, "From causality: ")
+    throw new GraphQLError(errors)
```

Comment:
Is this necessary? 

As I understand things, if a loader fails in any way it rejects the promise and then the error handling built up within MP takes it and processes it accordingly.

Reply:
the problem is that i am using `node-fetch` with a graphql endpoint-  the request succeeds with an `errors` array + empty `data`.

Reply:
however i could move the throwing code into the causality loader I wrote so it isn't just returning a fetch `Response` ... 🤔

Reply:
> the request succeeds with an errors array + empty data

This is idiomatic GraphQL, why break out of this pattern?

Reply:
@damassi I think the issue is that, the loader here will actually never fail, unless Causality is itself down. So if you're 'manually' calling a GraphQL endpoint in this way, I think you'd need to also 'manually' see if there are unrecoverable `errors` present in _that_ response, and bubble those up.

Reply:
I know this is resolved but just to follow up here I moved this error array handling into the loader itself so we are no longer dealing with a 'fetch' response